### PR TITLE
Coverity 1520485: logically dead code

### PR DESCRIPTION
**Pull request by paulidale**
This is a false positive, but changing the condition so the code appears live seems safer than removing the check.

Fixes Coverity 1520485

### CODE REVIEW

File: crypto/cmp/cmp_ctx.c [hunk 1]
```diff
@@ -919,12 +919,12 @@ int OSSL_CMP_CTX_set_option(OSSL_CMP_CTX *ctx, int opt, int val)
...
     case OSSL_CMP_OPT_MSG_TIMEOUT:
-        if (val < 0)
+        if (val <= 0)
             val = 0;
         ctx->msg_timeout = val;
         break;
     case OSSL_CMP_OPT_TOTAL_TIMEOUT:
-        if (val < 0)
+        if (val <= 0)
             val = 0;
```
Comment (DDvO) [inline crypto/cmp/cmp_ctx.c:928]: Thank you for taking care of this Coverity issue.

When looking at the code at the beginning of this function, it turns out that the case `val < 0` is already covered for both timeout settings by
```
    default:
        min_val = 0;
        break;
    }
    if (val < min_val) {
        ERR_raise(ERR_LIB_CMP, CMP_R_VALUE_TOO_SMALL);
        return 0;
    }
```
so the complaint by Coverity about dead code is actually right (thus not a false positive).

Please just remove the two occurrences of
```
if (val < 0)
    val = 0;
```

**Review by DDvO — CHANGES_REQUESTED** [review]

File: crypto/cmp/cmp_ctx.c (hunk 1 above)
Comment (paulidale) [reply crypto/cmp/cmp_ctx.c]: I figured you added the checks for clarity & sanity.  Happy to remove them.

**Review by hlandau — APPROVED** [review]

**Review by DDvO — APPROVED** [review]
Fine now.

**Comment by DDvO** [comment]
I just wanted to merge this, but then noticed that the 2nd part of the commit message
```
    Coverity 1520485: logically dead code
    
    This is a false positive, but changing the condition so the code appears live
    seems safer than removing the check.

```
should be updated.

**Review by DDvO — CHANGES_REQUESTED** [review]
Please update the commit message.

**Comment by t8m** [comment]
I'll update the message when merging.

**Comment by t8m** [comment]
Merged to master, 3.1, and 3.0 branches. Thank you.

**Pull request closed without merging**